### DISCUSSION TITLE
Toggleable column for boolean

### DISCUSSION THREAD

**Discussion by manusiakemos** [rappasoft/laravel-livewire-tables#1882] [Feature Requests]
maybe toggleable column for boolean would be nice. so when we use it shows toggle input to toggle true false value from column
```php
BooleanColumn::make('Active')->toggleAble() 
```

**Comment by lrljoe** [comment]
Just to validate, you want to have an option so that a "Click" on the BooleanColumn will change the "Active" field on the Model/Row?

**Reply by manusiakemos** [reply]
yes. and if possible wen can using switch button

**Reply by lrljoe** [reply]
So I'm adding in the core code for toggleable() for a Boolean Column, which will allow for defining a method to call "on-click", which will pass in the Row ID.

However - this won't be a "switch" type just yet, as that will be a chunk of code to get working in all the supported themes consistently: Boostrap-4, Bootstrap-5, and Tailwind

## Currently
For now, if you want a switch, then you could achieve this by using a "setView" on your Boolean Column.

## This isn't released yet
but will be later this weekend:


### Toggleable

You may call a defined public function, which should live within your Table Component, to allow "toggling" against your database:

```php
BooleanColumn::make('Active', 'status')
    ->toggleable('changeStatus'),
```

Then your "changeStatus" method may look like
```php
    public function changeStatus(int $id)
    {
        $item = $this->model::find($id);
        $item->status = !$item->status;
        $item->save();
    }
```

### Toggleable Confirmation Message

You may define a confirmation message prior to executing your toggleable() method.  The method will only be executed upon confirming.
```php
BooleanColumn::make('Active', 'status')
    ->confirmMessage('Are you sure that you want to change the status?')
    ->toggleable('changeStatus'),
```

Then your "changeStatus" method may look like
```php
    public function changeStatus(int $id)
    {
        $item = $this->model::find($id);
        $item->status = !$item->status;
        $item->save();
    }
```

**Reply by lrljoe** [reply]
The above is now in v3.4.15

While this doesn't give a switch button, it does expose the appropriate methods to set a method to toggle the Value used in a BooleanColumn on-click.

A "Switch" Boolean will come down the line, however as it needs to support BS4/BS5/TW, it's a bit more complex!

**Comment by manusiakemos** [comment]
okay. thank you

**Reply by lrljoe** [reply]
No problem, more features will come down the line for the BooleanColumn, I just don't want to introduce a "Switch" for one theme that won't be present in others, and getting it working in BS4/BS5/Tailwind2/Tailwind3 and the beta of Tailwind4 is proving challenging!

**Reply by manusiakemos** [reply]
not a problem, as long as i can use custom view with the power of tailwind and alpine js if i want to use switch
